File: engine/engine-model/src/main/scala/tools/packet/guild/GuildMemberOnline.scala
package tools.packet.guild

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class GuildMemberOnline(private var _guildId: Int, private var _characterId: Int, private var _online: Boolean) extends PacketInput {
  def guildId: Int = _guildId

  def characterId: Int = _characterId

  def online: Boolean = _online

  override def opcode(): SendOpcode = SendOpcode.GUILD_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/guild/GuildMemberOnline.java
package tools.packet.guild;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record GuildMemberOnline(Integer guildId, Integer characterId, Boolean online) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.GUILD_OPERATION;
   }
}